client/state/data-layer/handler-loading.js

```diff
@@ -0,0 +1,28 @@
+/** @format */
+/**
+ * Internal dependencies
+ */
+import { mergeHandlers } from 'state/action-watchers/utils';
+
+export const state = {
+	handlers: {},
+};
+
+const requiredHandlers = new Set();
+
+export const requireHandlers = ( ...requires ) => {
+	let nextHandlers = state.handlers;
+
+	requires.forEach( ( [ id, handlers ] ) => {
+		if ( requiredHandlers.has( id ) ) {
+			return;
+		}
+
+		requiredHandlers.add( id );
+		nextHandlers = mergeHandlers( nextHandlers, handlers );
+	} );
+
+	if ( state.handlers !== nextHandlers ) {
+		state.handlers = nextHandlers;
+	}
+};
```

Comment:
@coderkevin this looks a lot like what you are doing in the `extensions` directory. I'm not wanting to insult you by repeating it here, just wasn't sure how to pull in the code you wrote.

Reply:
Would this replace the code there? If so, I'm cool with it. I just don't want it in two places if at all possible.

Reply:
not unless you wanted. on that note:

 - do we deregister any reducers/handlers now?
 - if so, I think we could simplify this further with a `Map()` instead of a set. would you have interest in my tossing out an idea?

Reply:
maybe we can consider ripping out the de-registration?

Reply:
I'm okay with that as long as we're sure we won't need to ever deregister anything.